java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -0,0 +1,691 @@
+        buildCondition(cond.condition(), thenBlock, elseBlock);
+        break;
+      }
+      case VARIABLE:
+        currentBlock.elements.add(tree);
+        VariableTree variableTree = (VariableTree) tree;
+        if (variableTree.initializer() != null) {
+          build(variableTree.initializer());
+        }
+        break;
+      case MULTIPLY:
+      case DIVIDE:
+      case REMAINDER:
+      case PLUS:
+      case MINUS:
+      case LEFT_SHIFT:
+      case RIGHT_SHIFT:
+      case UNSIGNED_RIGHT_SHIFT:
+      case AND:
+      case XOR:
+      case OR:
+      case GREATER_THAN:
+      case GREATER_THAN_OR_EQUAL_TO:
+      case LESS_THAN:
+      case LESS_THAN_OR_EQUAL_TO:
+      case EQUAL_TO:
+      case NOT_EQUAL_TO:
+        BinaryExpressionTree binaryExpressionTree = (BinaryExpressionTree) tree;
+        currentBlock.elements.add(tree);
+        build(binaryExpressionTree.rightOperand());
+        build(binaryExpressionTree.leftOperand());
+        break;
+      case ASSIGNMENT:
+      case LEFT_SHIFT_ASSIGNMENT:
+      case RIGHT_SHIFT_ASSIGNMENT:
+      case AND_ASSIGNMENT:
+      case REMAINDER_ASSIGNMENT:
+      case UNSIGNED_RIGHT_SHIFT_ASSIGNMENT:
+      case OR_ASSIGNMENT:
+      case XOR_ASSIGNMENT:
+      case DIVIDE_ASSIGNMENT:
+      case MULTIPLY_ASSIGNMENT:
+      case PLUS_ASSIGNMENT:
+      case MINUS_ASSIGNMENT:
+        AssignmentExpressionTree assignmentExpressionTree = (AssignmentExpressionTree) tree;
+        currentBlock.elements.add(tree);
+        build(assignmentExpressionTree.variable());
+        build(assignmentExpressionTree.expression());
+        break;
+      case MEMBER_SELECT:
+        MemberSelectExpressionTree mse = (MemberSelectExpressionTree) tree;
+        currentBlock.elements.add(mse);
+        build(mse.expression());
+        break;
+      case CONDITIONAL_AND: {
+        BinaryExpressionTree e = (BinaryExpressionTree) tree;
+        // process RHS
+        Block falseBlock = currentBlock;
+        currentBlock = createBlock(falseBlock);
+        build(e.rightOperand());
+        Block trueBlock = currentBlock;
+        // process LHS
+        currentBlock = createBranch(e, trueBlock, falseBlock);
+        build(e.leftOperand());
+        break;
+      }
+      case CONDITIONAL_OR: {
+        BinaryExpressionTree e = (BinaryExpressionTree) tree;
+        // process RHS
+        Block trueBlock = currentBlock;
+        currentBlock = createBlock(trueBlock);
+        build(e.rightOperand());
+        Block falseBlock = currentBlock;
+        // process LHS
+        currentBlock = createBranch(e, trueBlock, falseBlock);
+        build(e.leftOperand());
+        break;
+      }
+      case LABELED_STATEMENT: {
+        LabeledStatementTree s = (LabeledStatementTree) tree;
+        build(s.statement());
+        currentBlock = createBlock(currentBlock);
+        labels.put(s.label().name(), currentBlock);
+        break;
+      }
+      case SWITCH_STATEMENT: {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested code block into a method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1199)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 9. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 28 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1151)